### ISSUE TITLE
Enforcing TCF2.0

### ISSUE THREAD

**Issue by bretg** [prebid/Prebid.js#4867]
## Description

Related to #4801 and #4747 

Prebid.org is neither data controller nor data processor, but we supply software to both types of entities. Therefore, our software should provide all the controls necessary for both types of entities to run their businesses.

Would like to try a new way of reviewing large documents like this. Here's a link to the Google Doc that anyone can make comments on:
https://docs.google.com/document/d/1fBRaodKifv1pYsWY3ia-9K96VHUjd8kKvxZlOsozm8E/edit#

Basically, this is a pretty significant change to Prebid.js -- it offers publishers the option to enforce GDPR in a robustly configurable way.

**Comment by pm-harshad-mane** [comment]
I think considering the deadline, we should support TCF2 as we supported TCF1 in the first phase and later in the next phase we can discuss and implement #4747 if needed.

**Comment by bretg** [comment]
After discussions last week, here's the proposed path-forward for Prebid and TCF:

Prebid.js

1. We will build a `deviceAccess` flag (#4747 ) that can be used by publishers as an interim way to initially enforce Purpose 1.
2. Read the TCF strings from the new API (#4801) 
3. Implement 'basic' Purpose 1 enforcement as described in the doc.
4. Implement the remainder of the purposes as described in the doc.

Prebid Server

1. Both PBS-Go and PBS-Java will build towards the 'full' enforcement as described in the doc.
2. Either or both teams may decide to phase in the functionality, e.g. at first only supporting Purpose 1 and/or 'basic' enforcement.

Prebid SDK

1. The SDK will accept a new parameter as described in the doc allowing the app developer to instruct Prebid whether it's ok to read the IDFA. Other than that, it's a passthrough.
